frontend/src/libs/components/app/app.tsx

```diff
@@ -9,15 +9,22 @@ import { actions as authActions } from '#slices/auth/auth.js';
 
 const App: React.FC = () => {
   const dispatch = useAppDispatch();
-  const { authenticatedUserDataStatus } = useAppSelector(({ auth }) => ({
-    authenticatedUserDataStatus: auth.authenticatedUserDataStatus,
-  }));
+  const { authenticatedUserDataStatus, surveyPreferencesDataStatus } =
+    useAppSelector(({ auth }) => ({
+      authenticatedUser: auth.authenticatedUser,
+      authenticatedUserDataStatus: auth.authenticatedUserDataStatus,
+      surveyPreferencesDataStatus: auth.surveyPreferencesDataStatus,
+    }));
 
   useEffect(() => {
     void dispatch(authActions.getAuthenticatedUser());
   }, [dispatch]);
 
-  if (authenticatedUserDataStatus === DataStatus.PENDING) {
+  if (
+    authenticatedUserDataStatus === DataStatus.IDLE ||
+    authenticatedUserDataStatus === DataStatus.PENDING ||
+    surveyPreferencesDataStatus === DataStatus.PENDING
```

Comment:
this should be in survey screen, not for the whole app
```suggestion
```

Reply:
Done